### PR TITLE
WIP: Configure AppVeyor to run automated test on Windows (#434)

### PR DESCRIPTION
**Pull request by jonas**
Configured here: https://ci.appveyor.com/project/jonas35151/akka-http

- [ ] Look into test failures: https://ci.appveyor.com/project/jonas35151/akka-http
- [ ] Ensure Ivy/sbt files are cached maybe this requires a successfully completed run
- [ ] What Scala version(s) to use, e.g. only compile for 2.11 or 2.12?
- [ ] Should we speed up AppVeyor build by annotating specific tests that should run on windows, e.g. tests for #346

### CODE REVIEW

**Comment by jonas** [comment]
Test failures
- ClientServerSpec: https://ci.appveyor.com/project/jonas35151/akka-http/build/3#L2029
```
[info]     - should close connection with idle client after idleTimeout *** FAILED ***
[info]       Expected exception akka.stream.StreamTcpException to be thrown, but java.util.concurrent.TimeoutException was thrown (ClientServerSpec.scala:243)```
```
- DontLeakActorsOnFailingConnectionSpecs: https://ci.appveyor.com/project/jonas35151/akka-http/build/3#L4731
```
[info] - should not leak connection Actors when hitting non-existing endpoint *** FAILED ***
[info]   java.lang.AssertionError: assertion failed: timeout (-98970700 nanoseconds) during expectMsgClass waiting for class akka.stream.impl.StreamSupervisor$Children
```

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+version: '{build}'
+os: Windows Server 2012
```
Comment (ktoso) [inline appveyor.yml]: is this the windows we want to run on? cc @drewhk

**Comment by ktoso** [comment]
Pretty cool, thanks. I guess we should set it up using some akka account so the address would be akka/akka-http?

Can we add multiple people to admin this appveyor account?

File: appveyor.yml (hunk 1 above)
Comment (drewhk) [inline appveyor.yml]: I have no clue, but I don't expect much differences.

**Comment by jonas** [comment]
To create an [organizational account](https://www.appveyor.com/docs/team-setup/#setting-up-appveyor-account-for-github-organization) my understanding is that you first create an account with GitHub signin then create an account using the normal signup from which you can add your GitHub linked accounts as an administrator.

Regarding having multiple people, yes it is possible. To quote the [Team Setup](https://www.appveyor.com/docs/team-setup/#introduction) page:
> AppVeyor Team page allows you adding new users or collaborators to your AppVeyor account as well as configure integration with GitHub teams.

**Comment by ktoso** [comment]
(Parked until after 10.0.0).
Very interesting to get this in though I'm somewhat scared it may be very unstable... no idea what boxes those are hm

**Comment by ktoso** [comment]
Do you mind if we closed this for now? Trying to clean up the PR queue only for PRs which we expect to be merged soon.

**Comment by jonas** [comment]
Yes, let's close it.

**Pull request closed without merging**